### PR TITLE
perf: Update the binding changes in parallel to improve performance when many clusters are picked.

### PR DESCRIPTION
**Pull request by ryanzhang-oss**
### Description of your changes

Update the binding changes in parallel to improve performance when many clusters are picked.

Fixes #

I have:

- [x] Run `make reviewable` to ensure this PR is ready for review.

### CODE REVIEW

**Review by michaelawyu — APPROVED** [review]
LGTM.

File: pkg/scheduler/framework/framework.go [hunk 1]
```diff
@@ -662,26 +663,53 @@ func (f *framework) manipulateBindings(
...
 	for _, bp := range toPatch {
-		// Use JSON patch to avoid races.
-		if err := f.client.Patch(ctx, bp.updated, bp.patch); err != nil {
-			return controller.NewUpdateIgnoreConflictError(fmt.Errorf("failed to patch binding %s: %w", bp.updated.Name, err))
-		}
+		patchBinding := bp
+		errs.Go(func() error {
+			return retry.OnError(retry.DefaultBackoff,
+				func(err error) bool {
+					return apierrors.IsServiceUnavailable(err) || apierrors.IsServerTimeout(err)
```
Comment (zhiying-lin) [inline pkg/scheduler/framework/framework.go:700]: do we need to retry when it conflicts?

File: pkg/scheduler/framework/framework.go [hunk 2]
```diff
@@ -662,26 +663,53 @@ func (f *framework) manipulateBindings(
...
+							return nil
+						}
+						klog.ErrorS(err, "failed to create a new binding", "clusterResourceBinding", klog.KObj(newBinding))
+					}
+					return err
+				})
+		})
 	}
-	return nil
+	return errs.Wait()
```
Comment (zhiying-lin) [inline pkg/scheduler/framework/framework.go:688]: wrapper the error, NewAPIServerError()?

File: pkg/scheduler/framework/framework.go [hunk 3]
```diff
@@ -662,26 +663,53 @@ func (f *framework) manipulateBindings(
...
+					err := f.client.Patch(cctx, patchBinding.updated, patchBinding.patch)
+					if err != nil {
+						klog.ErrorS(err, "failed to patch a binding", "clusterResourceBinding", klog.KObj(patchBinding.updated))
+					}
+					return err
+				})
+		})
 	}
-	return nil
+	return errs.Wait()
```
Comment (zhiying-lin) [inline pkg/scheduler/framework/framework.go:712]: wrapper the error, NewAPIServerError()?

File: pkg/scheduler/framework/framework.go [hunk 4]
```diff
@@ -662,26 +663,53 @@ func (f *framework) manipulateBindings(
...
+					return apierrors.IsServiceUnavailable(err) || apierrors.IsServerTimeout(err)
+				},
+				func() error {
+					err := f.client.Create(cctx, newBinding)
+					if err != nil {
+						if apierrors.IsAlreadyExists(err) {
+							// The binding already exists, which is fine.
+							return nil
+						}
+						klog.ErrorS(err, "failed to create a new binding", "clusterResourceBinding", klog.KObj(newBinding))
```
Comment (zhiying-lin) [inline pkg/scheduler/framework/framework.go:682]: klog.ErrorS(err, "Failed to create a new binding", "clusterResourceBinding", klog.KObj(newBinding))

log with upper case

File: pkg/scheduler/framework/framework.go [hunk 5]
```diff
@@ -662,26 +663,53 @@ func (f *framework) manipulateBindings(
...
+		errs.Go(func() error {
+			return retry.OnError(retry.DefaultBackoff,
+				func(err error) bool {
+					return apierrors.IsServiceUnavailable(err) || apierrors.IsServerTimeout(err)
+				},
+				func() error {
+					// Use JSON patch to avoid races so we shouldn't get a conflict error.
+					err := f.client.Patch(cctx, patchBinding.updated, patchBinding.patch)
+					if err != nil {
+						klog.ErrorS(err, "failed to patch a binding", "clusterResourceBinding", klog.KObj(patchBinding.updated))
```
Comment (zhiying-lin) [inline pkg/scheduler/framework/framework.go:706]: klog.ErrorS(err, "Failed to patch a binding", "clusterResourceBinding", klog.KObj(patchBinding.updated))

**Pull request merged**